app/controllers/admin/rewards_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Admin
+  class RewardsController < AdminController
+    def index
+      render :index, locals: { rewards: Reward.all }
+    end
+
+    def new
+      render :new, locals: { reward: Reward.new }
+    end
+
+    def edit
+      render :edit, locals: { reward: reward }
+    end
+
+    def create
+      reward = Reward.new(reward_params)
+      if reward.save
+        redirect_to admin_rewards_path(reward), notice: 'Reward was successfully created.'
+      else
+        render :new, locals: { reward: reward }
+      end
+    end
+
+    def update
+      if reward.update(reward_params)
+        redirect_to admin_rewards_path(reward), notice: 'Reward was successfully updated.'
+      else
+        render :edit, locals: { reward: reward }
+      end
+    end
+
+    def destroy
+      reward.destroy
+      redirect_to admin_rewards_url, notice: 'Reward was successfully destroyed.'
+    end
+
+    private
+
+    def reward
+      @reward = Reward.find(params[:id])
```

Comment:
It looks like I was wrong couple of PR ago - we can get something out of memoization here, since `reward` is called two times in a row in the `update` method - what do you think about it?

Reply:
exactly - apologies for the confusion :)

Reply:
I add a memorization to the reward :)